src/components/POLine/Item/ItemForm.js

```diff
@@ -88,64 +122,111 @@ class ItemForm extends Component {
     }
 
     this.setState(({
-      instanceId: inventoryData.instanceId,
-      title: get(inventoryData, 'titleOrPackage', ''),
-      publisher: get(inventoryData, 'publisher', ''),
-      publicationDate: get(inventoryData, 'publicationDate', null),
-      edition: get(inventoryData, 'edition', ''),
-      contributors: get(inventoryData, 'contributors', []),
-      productIds: get(inventoryData, 'productIds', []),
+      inventoryData: {
+        instanceId: inventoryData.instanceId,
+        title: get(inventoryData, 'titleOrPackage', ''),
+        publisher: get(inventoryData, 'publisher', ''),
+        publicationDate: get(inventoryData, 'publicationDate', null),
+        edition: get(inventoryData, 'edition', ''),
+        contributors: get(inventoryData, 'contributors', []),
+        productIds: get(inventoryData, 'productIds', []),
+      },
     }));
   };
 
   onChangeField = (value, fieldName) => {
+    clearTimeout(this.changeItemDetailsTimeout);
+
     const { change, formValues } = this.props;
-    const inventoryData = this.state;
-    const locations = formValues?.locations;
+    const { inventoryData } = this.state;
+
+    const connectedInstance = formValues?.instanceId;
+    const rollbackData = {
+      isPackage: formValues.isPackage,
+      ...getNormalizedInventoryData(inventoryData),
+    };
+    const rollbackValue = get(rollbackData, fieldName);
 
     if (fieldName) change(fieldName, value);
 
-    setTimeout(() => {
-      if (shouldSetInstanceId(this.props.formValues, inventoryData)) {
-        change('instanceId', inventoryData.instanceId);
-      } else {
-        change('instanceId', null);
-        locations?.forEach((_, i) => change(`locations[${i}].holdingId`, null));
-      }
+    return new Promise((resolve, reject) => {
+      this.changeItemDetailsTimeout = setTimeout(() => {
+        if (shouldSetInstanceId(this.props.formValues, inventoryData)) {
+          change('instanceId', inventoryData.instanceId);
+          resolve();
+        } else if (connectedInstance) {
+          this.onBreakInstanceConnection()
+            .then(resolve)
+            .catch(() => {
+              change(fieldName, rollbackValue);
+              reject();
+            });
+        } else {
+          this.breakInstanceConnection();
+          resolve();
+        }
+      });
     });
   };
 
+  safeChangeField = (value, fieldName) => {
+    return this.onChangeField(value, fieldName).catch(noop);
```

Comment:
What is the reason to have useless `.catch`?

Reply:
`onChangeField` rejects when the field should not eventually be updated (e.g. user cancel instance connection break). This is used to interrupt sequential change of dependent fields (e.g. 'isPackage' -> 'checkinItems') and some fields don't really care about handling the rejection, so the `catch` uses `noop` in this case just to not throw an unhandled exception.